src/saveload/group_sl.cpp

```diff
@@ -59,6 +60,13 @@ struct GRPSChunkHandler : ChunkHandler {
 				g->livery.colour1 = c->livery[LS_DEFAULT].colour1;
 				g->livery.colour2 = c->livery[LS_DEFAULT].colour2;
 			}
+
+			Company *c = Company::Get(g->owner);
+			if (IsSavegameVersionBefore(SLV_GROUP_NUMBERS)) {
+				g->number = c->freegroups.UseID(c->freegroups.NextID());
```

Comment:
I'm not sure renumbering existing groups is the right choice. I can't think of a reason that players would use automatically-numbered groups, but I've learned not to assume I would know. 🙂 

Maybe it would be better to keep the existing number?

```suggestion
				g->number = c->freegroups.UseID(g->index);
```

Reply:
If you want to keep the group number then `g->index` is the wrong thing to use.

But also, while groups themselves don't care about unique names, the ID generator doesn't have any idea what to do with duplicated numbers and will not do the right thing (and would be inconsistent in multiplayer.)

Reply:
Ah, well then, changing numbers is probably fine.

Reply:
You are correct, I was thinking about completely the wrong thing -- taking over a company, which deliberately renumbers just it does for unit numbers.

Sorry!

So, yes, it is feasible to use `g->index` during savegame conversion.